packages/npm-dependencies-validation/src/utils/npmUtil.ts

```diff
@@ -0,0 +1,26 @@
+import { spawn } from "child_process";
+
+export function getNPM(): string {
+  return /^win/.test(process.platform) ? "npm.cmd" : "npm";
+}
+
+export function spawnCommand(
+  commandArgs: string[],
+  workingDir: string
+): Promise<any> {
+  return new Promise((resolve, reject) => {
+    const command = spawn(getNPM(), [...commandArgs, "--json"], {
+      cwd: workingDir,
+    });
+
+    command.stdout.on("data", (data) => {
```

Comment:
is there any max size buffer here?
does all the JSON gets passed in a **single** `data` event?
Please verify using the node docs

Reply:
By default, pipes for stdin, stdout, and stderr are established between the parent Node.js process and the spawned subprocess. These pipes have limited (and platform-specific) capacity. If the subprocess writes to stdout in excess of that limit without the output being captured, the subprocess blocks waiting for the pipe buffer to accept more data.